packages/api/src/promise/SubmittableExtrinsic.ts

```diff
@@ -22,17 +23,40 @@ export default class SubmittableExtrinsic extends Extrinsic {
     this._api = api;
   }
 
+  private checkStatus (statusCb: (result: SendResult) => any): (status: ExtrinsicStatus) => Promise<void> {
+    return async (status: ExtrinsicStatus): Promise<void> => {
+      let events: Array<any> | undefined = undefined;
+
+      if (status.type === 'Finalised') {
+        const blockHash = status.value as Hash;
+        const { block: { extrinsics } }: SignedBlock = await this._api.rpc.chain.getBlock(blockHash);
+        const allEvents: Array<EventRecord> = await this._api.query.system.events.at(blockHash) as any;
+
+        const index = extrinsics
+          .map((ext) => ext.hash.toHex())
+          .indexOf(this.hash.toHex());
+
+        if (index !== -1) {
+          events = allEvents.filter(({ phase }) =>
+            phase.type === 'ApplyExtrinsic' && (phase.value as u32).eqn(index)
```

Comment:
Why not only `(phase.value as u32).eqn(index)`

Reply:
in `Finalisation` we don't have the value, e.g. https://github.com/polkadot-js/api/blob/master/packages/types/src/EventRecord.ts#L24 it is only on the `ApplyExtrinsic`